### DISCUSSION TITLE
How can skywalking be deployed in different roles?

### DISCUSSION THREAD

**Discussion by cuixianyu-github** [apache/skywalking#10730] [Q&A] [answered]
Could you please tell me how to deploy skywalking by role? Do you only need to modify the role in application.xml? Who has the relevant document link to send me next, thank you

**Comment by wu-sheng** [comment]
Could you explain why do you need this in k8s? Generally, OAP cluster in K8s should be `mixed` mode automatically. 
Ref doc, https://skywalking.apache.org/docs/main/v9.4.0/en/setup/backend/backend-cluster/#kubernetes

**Comment by wu-sheng** [comment, marked as answer]
If you have to do that, you usually have to set up two OAP deployments from the k8s concept, but share the same label values for `cluster/kubernetes/labelSelector`.
We don't support this if you choose SkyWalking's official helm or operator, because generally, we can't find out the point of this kind of deployment. 

## Notice
**These roles are designed for complex deployment requirements on security and network policy.**

This is mentioned in the docs, ref https://skywalking.apache.org/docs/main/v9.4.0/en/setup/backend/advanced-deployment/

In the k8s, network between pods are managed by the k8s itself. We should only expect it is manually designed.
